_docs/intro/overview.md

```diff
@@ -4,31 +4,31 @@ title: Overview
 
 TeaVM is an ahead-of-time compiler of Java bytecode to JavaScript.
 It's much like GWT, however GWT takes source code, and this limits GWT to Java only.
-Unlike GWT, TeaVM relies on existing compilers, be it javac, kotlinc or scalac.
+Unlike GWT, TeaVM relies on existing compilers such as javac, kotlinc and scalac.
 These compilers produce bytecode (`*.class` or `*.jar` files),
 then TeaVM takes this bytecode and produces JavaScript code.
 
 
 # Purpose
 
-TeaVM is a primarily web development tool.
-It's not for getting your large existing codebase in Java or Kotlin and producing JavaScript.
+TeaVM is primarily a web development tool.
+It's not for taking your large existing codebase in Java or Kotlin and producing JavaScript.
 Unfortunately, Java was not designed to run efficiently in the browser.
 There are Java APIs that are impossible to implement without generating inefficient JavaScript.
-Some of these APIs are: reflection, resources, class loaders, JNI.
-TeaVM restricts usages of these APIs.
+Some of these APIs are: reflection, resources, class loaders, and JNI.
+TeaVM restricts usage of these APIs.
 Generally, you'll have to manually rewrite your code to fit into TeaVM constraints.
 
 TeaVM is for you, if:
 
-  * You are a Java developer and you are going to write web front-end from scratch.
-  * You already have Java-based backend and want to integrate front-end code tightly into your existing development
+  * You are a Java developer and you are going to write a web front-end from scratch.
+  * You already have a Java-based backend and want to integrate front-end code tightly into your existing development
     infrastructure.
-  * You have some Java back-end code you want to reuse in front-end.  
+  * You have some Java back-end code you want to reuse in the front end.  
   * You are ready to rewrite your code to work with TeaVM.
 
-If you have bloated applications that use Swing, you want to run these applications in web,
-and you don't care about download size, start-up time and performance, you better give up;
+If you have tightly-coupled applications that use Swing, you want to run these applications in web,
```

Comment:
This changes the meaning. 'Bloated' means 'large', 'immense', not poorly engineered (which I believe referred by term 'tightly-coupled').

Reply:
"Bloated" does mean those things, but it has negative connotations -- a Swing fan who reads this might think you are disparaging all applications that use Swing.  I realize that may not be what you mean, but it's what it sounds like upon first reading.

The reason I said "tightly-coupled" is that a well-engineered Swing application that separates business logic and UI cleanly _could_ be adapted to work with TeaVM -- just add another 'teavm' maven module atop the business logic, in parallel with the 'swing' module.  By contrast, a tightly-coupled application with business logic intermingled directly with UI code would be very difficult to convert to TeaVM, it would probably be easier to just rewrite it.

So I was trying to change the wording to keep your idea (which I agree with), while not offending Swing developers who would probably be a great target audience for TeaVM, as they look to move to the web.